numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3687,6 +3694,19 @@ static void
 #define _HALF_LESS_THAN(a, b) (!npy_half_isnan(a) && npy_half_lt_nonan(a, b))
 #define _HALF_GREATER_THAN(a, b) (!npy_half_isnan(a) && npy_half_lt_nonan(b, a))
 
+/* The following function provides Optimization for INT_fastclip using avx2 extension.
+First it checks if AVX2 attribute is present or not.If AVX2 support is present, then it
+breaks down the Algorithm into two parts.
+1. First it looks at total length of the input array and find out how many bytes are unaligned
+   with respect to 256 bytes. Because avx2 registers used here are 256 bytes in size.
+2. Then for the portion it is aligned with 256 bytes, it loops over the data and loads into one of the vector registers.
+3. It loads vector registers vec_max_256 as vector of maximum value and vec_min_max as vector of minimum values, each of 4 Bytes.
+4. Parallel comparision is made between loaded data and vec_max_256 first and do a vectorized min operation betwwen data and max value. If data is greater than max value , it gets replaced by max value, otherwise with data.
+5. We then use the register where result of above min comparison is tored and do a max operation between the data and min value. If data is lesser than the min value , it gets replaced by min value, otherwise with data. 
+6. After that we take care of the unaligned portion by simply looping over it, like the previous algorithm was doing.
+7. For a simple benchmark, this algorithm gave around 40-50% performance boost.
+*/
```

Comment:
I think most of these comments would be better inline with the bit of code that does them